### PR TITLE
Added Basic Step

### PR DESCRIPTION
**Pull request by GelaPT**

### CODE REVIEW

File: code/Player/Controllers/HumanController.cs [hunk 1]
```diff
@@ -147,6 +150,41 @@ private void Walk()
...
+			
+			ClimbStep();
+		}
+
+		/// <summary>
+		/// Tries to climb steps in case player can't just simply walk.
+		/// </summary>
+		private void ClimbStep() {
+			MoveHelper mover = CheckMove();
+			if (mover.TryMove(Time.Delta) > 0.9) return;
```
Comment (Prakkkmak) [inline code/Player/Controllers/HumanController.cs:164]: magic number (0.9)

File: code/Player/Controllers/HumanController.cs [hunk 2]
```diff
@@ -147,6 +150,41 @@ private void Walk()
...
+				.Ignore(Pawn)
+				.Run();
+
+			Position = ray.EndPos;
+		}
+
+		/// <summary>
+		/// Try to do simple walk
+		/// </summary>
+		/// <returns>Returns if simple walk was possible</returns>
```
Comment (Prakkkmak) [inline code/Player/Controllers/HumanController.cs:180]: It returns a MoveHelper, not an "if simple walk was possible" 
Basicaly the user of this method can't really understind what to do with the return

File: code/Player/Controllers/HumanController.cs [hunk 3]
```diff
@@ -1,4 +1,6 @@
 ﻿using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
```
Comment (Prakkkmak) [inline code/Player/Controllers/HumanController.cs:3]: Why this import ?

File: code/Player/Controllers/HumanController.cs [hunk 4]
```diff
@@ -83,17 +89,34 @@ protected virtual void TryPlayerMove()
...
 			Position = mover.Position;
 			Velocity = mover.Velocity;
 		}
+		
+		/// <summary>
+		/// Moves the player in the direction of their velocity and if there is colliding objects, checks if it can step.
+		/// </summary>
+		protected virtual void TryPlayerMoveWithStep()
+		{
+			if (Velocity.Length <= 1.0F)
```
Comment (Prakkkmak) [inline code/Player/Controllers/HumanController.cs:106]: magic number

File: code/Player/Controllers/HumanController.cs [hunk 5]
```diff
@@ -103,6 +126,14 @@ protected virtual void UpdateBBox()
 			maxs = new Vector3( +BodyGirth, +BodyGirth, BodyHeight ) * Pawn.Scale;
```
Comment (Prakkkmak) [inline code/Player/Controllers/HumanController.cs:126]: Not your modification but somebody know why the '+' ?

**Review by Prakkkmak — CHANGES_REQUESTED** [review]

File: code/Player/Controllers/HumanController.cs (hunk 4 above)
Comment (GelaPT) [reply code/Player/Controllers/HumanController.cs:106]: it's for ignoring small movements, it's nauwp's idea

Comment (Prakkkmak) [reply code/Player/Controllers/HumanController.cs:106]: Yeah i know. Magic number means: all constant have to be declared outside the scope. In this example you have to declare and init this value in the properties of the class as const

**Review by Prakkkmak — APPROVED** [review]

File: code/Player/Controllers/HumanController.cs (hunk 5 above)
Comment (Nowp) [reply code/Player/Controllers/HumanController.cs:125]: To be explicit about the sign I think

File: code/Player/Controllers/HumanController.cs [hunk 6]
```diff
@@ -103,6 +125,14 @@ protected virtual void UpdateBBox()
...
 		}
 
+		private MoveHelper GetMoveHelper() {
+			MoveHelper mover = new(Position, Velocity);
+			mover.Trace = mover.Trace.Size(mins, maxs).Ignore(Pawn);
+			mover.MaxStandableAngle = GroundAngle;
+
+			return mover;
+		}
```
Comment (Nowp) [inline code/Player/Controllers/HumanController.cs:134]:
```suggestion
		private MoveHelper GetMoveHelper() 
		{
			MoveHelper mover = new(Position, Velocity);
			mover.Trace = mover.Trace.Size(mins, maxs).Ignore(Pawn);
			mover.MaxStandableAngle = GroundAngle;

			return mover;
		}
```

Comment (Prakkkmak) [reply code/Player/Controllers/HumanController.cs:134]: What did you changed ? :o

File: code/Player/Controllers/HumanController.cs [hunk 7]
```diff
@@ -103,6 +125,14 @@ protected virtual void UpdateBBox()
 			maxs = new Vector3( +BodyGirth, +BodyGirth, BodyHeight ) * Pawn.Scale;
 		}
 
+		private MoveHelper GetMoveHelper() {
```
Comment (Prakkkmak) [inline code/Player/Controllers/HumanController.cs:128]:
```suggestion
		private MoveHelper GetMoveHelper() 
		{
```

File: code/Player/Controllers/HumanController.cs (hunk 6 above)
Comment (Nowp) [reply code/Player/Controllers/HumanController.cs:134]: The opening bracket

File: code/Player/Controllers/HumanController.cs (hunk 7 above)
Comment (Nowp) [reply code/Player/Controllers/HumanController.cs:128]: I resolve your comment since there's already my comment opened

**Comment by Prakkkmak** [comment]
[Climb Step](https://trello.com/c/bnLiUwGf/4-climb-step)

**Review by Nowp — APPROVED** [review]

**Pull request merged**